### PR TITLE
Jamiecook/bebop

### PR DESCRIPTION
**Pull request by jamiecook**

### CODE REVIEW

File: linux/build_odb_thing.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    if thing == "debug":
+        cmd += "config.cc.coptions=-g "
+    run(cmd)
+    return build_dir
+
+    with chdir(build_dir): 
+        print(os.getcwd())
+        run("bpkg build odb@https://pkg.cppget.org/1/beta --yes --trust-yes")
+        run("bpkg test odb")
+        run("bpkg install odb")
```
Comment (jamiecook) [inline linux/build_odb_thing.py]:
```suggestion
```

File: linux/build_odb_thing.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            # run("bpkg build libodb-pgsql")
+            # run("bpkg build libodb-mysql")
+            run("bpkg install --all --recursive")
+        else:
+            raise RuntimeError(f"Don't know how build odb-thing: {thing}")
+    
+
+def create_bpkg_build_dir(deps_dir, version, thing):
+    # os.environ['PATH'] = "/home/jamie/git/polaris-dependencies/tmp/gcc-9.4.0/build2-build/tmp/gcc-9.4.0/build2/bin" + os.pathsep + os.environ['PATH']
+
```
Comment (jamiecook) [inline linux/build_odb_thing.py]:
```suggestion
```

File: linux/build_odb_thing.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    install_dir = join(deps_dir, f"odb-{version}-{thing}")
+    build_dir = join(deps_dir, f"build-odb-{version}-{thing}")
+    lib_dir=join(install_dir, 'lib')
+    mkdir_p(lib_dir)
+    mkdir_p(build_dir)
+
+
+    opt_level="-O3"
+    if thing == "debug":
+        opt_level = "-O0"
```
Comment (jamiecook) [inline linux/build_odb_thing.py]:
```suggestion
    opt_level = "-O3" if thing == "debug" else "-O0"
```

File: linux/build_odb_thing.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+# cd $build_folder
+# bpkg build odb@https://pkg.cppget.org/1/beta --yes --trust-yes
+# bpkg test odb
+# bpkg install odb 
+
+
+
+# if __name__ == "__main__":
+#     main()
```
Comment (jamiecook) [inline linux/build_odb_thing.py]:
```suggestion
```

File: linux/build_odb_thing.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(prev_cwd)
+
+def build_odb(deps_dir, version):
+     #deps_dir = abspath(sys.argv[1])
+     #version = sys.argv[2]
+     #thing = sys.argv[3]
```
Comment (jamiecook) [inline linux/build_odb_thing.py]:
```suggestion
```

File: linux/build-odb-compiler.sh [hunk 6]
```diff
@@ -6,21 +6,23 @@ set -eu
...
-gcc_ver=$($filedir/../gcc_ver.sh $CXX)
+gcc_ver=$(sh $filedir/../gcc_ver.sh $CXX)
 
 mkdir -p $gcc_ver-compiler
 
 install_root=$1
 install_path=$install_root/odb-2.5.0-compiler
 lib_path=$install_path/lib
 build_folder=build-odb-2.5.0-$gcc_ver-compiler
+plugin_dir=/blues/gpfs/software/centos7/spack-latest/opt/spack/linux-centos7-x86_64/gcc-6.5.0/gcc-10.2.0-z53hda3/lib/gcc/x86_64-pc-linux-gnu/10.2.0/plugin/include
```
Comment (jamiecook) [inline linux/build-odb-compiler.sh:18]:
```suggestion
# The following is sometimes necessary for compiling on bebop
plugin_dir=/blues/gpfs/software/centos7/spack-latest/opt/spack/linux-centos7-x86_64/gcc-6.5.0/gcc-10.2.0-z53hda3/lib/gcc/x86_64-pc-linux-gnu/10.2.0/plugin/include
```

File: linux/build-odb-compiler.sh [hunk 7]
```diff
@@ -6,21 +6,23 @@ set -eu
...
 #BPKG_BIN=$install_root/build2/bin/bpkg
 mkdir -p $install_path
 cd $install_root
-bpkg create -d $build_folder cc  --wipe       \
+bpkg create -d $build_folder cc  --wipe         \
   config.cxx=$CXX                  		\
   config.cc.coptions=-O3          		\
   config.cxx.coptions=-std=c++17	  	\
+  config.cxx.coptions=-I${plugin_dir}	   	\
```
Comment (jamiecook) [inline linux/build-odb-compiler.sh]:
```suggestion
  # config.cxx.coptions=-I${plugin_dir}	   	\
```

**Review by whiteg-anl — APPROVED** [review]
Haven't tested on Bebop yet but the linux build changes all work

**Pull request merged**